modules/sirdataRtdProvider.js

```diff
@@ -0,0 +1,423 @@
+    // if children or "Do not Sell" management in California, no segments, page categories only whatever TCF signal
+    sirdataDomain = 'cookieless-data.com';
+    sendWithCredentials = false;
+    gdprApplies = null;
+    tcString = '';
+  } else if (getGlobal().getConfig('consentManagement.gdpr')) {
+  // Default endpoint is cookieless if gdpr management is set. Needed because the cookie-based endpoint will fail and return error if user is located in Europe and no consent has been given
+    sirdataDomain = 'cookieless-data.com';
+    sendWithCredentials = false;
+  }
+
+  // default global endpoint is cookie-based if no rules falls into cookieless or consent has been given or GDPR doesn't apply
+  if (!sirdataDomain || !gdprApplies || (hasOwnDeepProperty(userConsent, 'consents') && userConsent.gdpr.vendorData.vendor.consents[53] && userConsent.gdpr.vendorData.purpose.consents[1] && userConsent.gdpr.vendorData.purpose.consents[4])) {
+    sirdataDomain = 'sddan.com';
+    sendWithCredentials = true;
+  }
+
+  var actualUrl = moduleConfig.params.actualUrl || getRefererInfo().referer;
+
+  const url = 'https://kvt.' + sirdataDomain + '/api/v1/public/p/' + moduleConfig.params.partnerId + '/d/' + moduleConfig.params.key + '/s?callback=&gdpr=' + gdprApplies + '&gdpr_consent=' + tcString + (actualUrl ? '&url=' + actualUrl : '');
+  ajax(url, {
+    success: function (response, req) {
+      if (req.status === 200) {
+        try {
+          const data = JSON.parse(response);
+          if (data && data.segments) {
+            addSegmentData(adUnits, data, moduleConfig, onDone);
+          } else {
+            onDone();
+          }
+        } catch (e) {
+          onDone();
+          utils.logError('unable to parse Sirdata data' + e);
+        }
+      } else if (req.status === 204) {
+        onDone();
+      }
+    },
+    error: function () {
+      onDone();
+      utils.logError('unable to get Sirdata data');
+    }
+  },
+  null,
+  {
+    contentType: 'text/plain',
+    method: 'GET',
+    withCredentials: sendWithCredentials,
+    referrerPolicy: 'unsafe-url',
+    crossOrigin: true
+  });
+}
+
+export function setGlobalOrtb2(segments, categories) {
+  try {
+    let addOrtb2 = {};
+    _set(addOrtb2, 'user.ext.data.sd_rtd', segments || {}, false);
+    _set(addOrtb2, 'site.ext.data.sd_rtd', categories || {}, false);
+    let ortb2 = {ortb2: utils.mergeDeep({}, getGlobal().getConfig('ortb2') || {}, addOrtb2)};
+    getGlobal().setConfig(ortb2);
+  } catch (e) {
+    utils.logError(e)
+  }
+
+  return true;
+}
+
+export function setBidderOrtb2(bidder, segments, categories) {
+  try {
+    let addOrtb2 = {};
+    _set(addOrtb2, 'user.ext.data.sd_rtd', segments || {}, false);
+    _set(addOrtb2, 'site.ext.data.sd_rtd', categories || {}, false);
+    let testBidder = config.getBidderConfig(bidder);
+    let ortb2 = {ortb2: utils.mergeDeep({}, ((testBidder.rubicon && testBidder.rubicon.ortb2) ? testBidder.rubicon.ortb2 : {}), addOrtb2)};
```

Comment:
Is there a reason that rubicon is called out as a bidder here? Should it not be:

`(testBidder[bidder] && testBidder[bidder].ortb2) ? testBidder[bidder].ortb2 : {}`

Reply:
OMG thank you very much for your vigilance! Off course you're right and fix has been pushed...